biz.aQute.bndlib/src/aQute/bnd/osgi/Verifier.java

```diff
@@ -673,7 +674,12 @@ public void verify() throws Exception {
 		verifyDirectives(Constants.IMPORT_PACKAGE, "resolution:", PACKAGEPATTERN, "package");
 		verifyDirectives(Constants.REQUIRE_BUNDLE, "visibility:|resolution:", SYMBOLICNAME, "bsn");
 		verifyDirectives(Constants.FRAGMENT_HOST, "extension:", SYMBOLICNAME, "bsn");
-		verifyDirectives(Constants.PROVIDE_CAPABILITY, "effective:|uses:", null, null);
+		verifyDirectives(Constants.PROVIDE_CAPABILITY, "effective:|uses:", null, null, (namespace, directive) -> {
```

Comment:
Perhaps a better way, is for the lamba return of the directives string used to make the pattern? Then the default is to return "effective:|uses:" but the lambda could return "effective:|uses:|register:" for the osgi.serviceloader namespace? This way we don't need special code, just customized directives strings based upon the namespace.

BTW, great idea for the solution!

Reply:
ok, you mean a simple `Function<String, String>`  input is the namespace, return is the directives pattern?